pyproject.toml

```diff
@@ -13,7 +13,7 @@ license = "Apache 2.0"
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = ">=3.7.9,<4"
+python = ">=3.8.18,<4"
```

Comment:
specifying a minor version at the lower bound is unusual - why not `>=3.8`?

(the upper bound is also controversial, at least in some circles.  Since there is no imminent prospect of python 4 it probably does not matter, but just `python = ">=3.8"` as the entire constraint should be fine).

I also notice that there are quite a few outdated dev dependencies here.  My preference for dev dependencies is `>=` constraints rather than caret or specific versions, your mileage may vary

Reply:
also you forgot to update poetry.lock